README.md

```diff
@@ -1,3 +1,61 @@
 # KubeSaw Docs
 
 This repository contains the documentation website code and Markdown source files for kubesaw.github.io.
+
+## Contributing
+
+The website is built using [mkdocs](https://www.mkdocs.org/getting-started/#getting-started-with-mkdocs)
+
+**NOTE**: To contribute changes, make sure you do that from your own fork!
+
+### Prerequisites
+
+- python3
+```shell
+python3 --version
+Python 3.11.5
+```
+
+### 1.  Create an isolated virtual environment
+From the root folder of the repo run: 
+```shell
+python3 -m venv venv
+``` 
+NOTE: the second `venv` is the name of the folder containing the [virtual environment](https://docs.python.org/3/library/venv.html)
+
+Now you need to activate the virtual env so that all the dependencies will be installed only here, not on your global python installation.
+
+```shell
+source venv/bin/activate
+```
+
+### 2. Install dependencies
+
+```shell
+pip install -r requirements.txt 
+```
+
+NOTE:
+- add newly needed dependencies in the `requirements.txt` file
+
+### 3. Run the local server
+
+```shell
+mkdocs serve
```

Comment:
this failed for me with:
```
Error: Config file 'mkdocs.yml' does not exist.
```
not sure what I'm missing now :thinking: 

```
/[~/kubesaw/kubesaw.github.io] ((06c2bc5...))$ python3 --version
Python 3.11.8
/[~/kubesaw/kubesaw.github.io] ((06c2bc5...))$ python3 -m venv venv
/[~/kubesaw/kubesaw.github.io] ((06c2bc5...))$ source venv/bin/activate
/[~/kubesaw/kubesaw.github.io] ((06c2bc5...))$ pip install -r requirements.txt 
Requirement already satisfied: mkdocs in ./venv/lib64/python3.11/site-packages (from -r requirements.txt (line 1)) (1.6.1)
Requirement already satisfied: mkdocs-material in ./venv/lib64/python3.11/site-packages (from -r requirements.txt (line 2)) (9.5.47)
Requirement already satisfied: click>=7.0 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (8.1.7)
Requirement already satisfied: ghp-import>=1.0 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (2.1.0)
Requirement already satisfied: jinja2>=2.11.1 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (3.1.4)
Requirement already satisfied: markdown>=3.3.6 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (3.7)
Requirement already satisfied: markupsafe>=2.0.1 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (3.0.2)
Requirement already satisfied: mergedeep>=1.3.4 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (1.3.4)
Requirement already satisfied: mkdocs-get-deps>=0.2.0 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (0.2.0)
Requirement already satisfied: packaging>=20.5 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (24.2)
Requirement already satisfied: pathspec>=0.11.1 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (0.12.1)
Requirement already satisfied: pyyaml-env-tag>=0.1 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (0.1)
Requirement already satisfied: pyyaml>=5.1 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (6.0.2)
Requirement already satisfied: watchdog>=2.0 in ./venv/lib64/python3.11/site-packages (from mkdocs->-r requirements.txt (line 1)) (6.0.0)
Requirement already satisfied: babel~=2.10 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (2.16.0)
Requirement already satisfied: colorama~=0.4 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (0.4.6)
Requirement already satisfied: mkdocs-material-extensions~=1.3 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (1.3.1)
Requirement already satisfied: paginate~=0.5 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (0.5.7)
Requirement already satisfied: pygments~=2.16 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (2.18.0)
Requirement already satisfied: pymdown-extensions~=10.2 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (10.12)
Requirement already satisfied: regex>=2022.4 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (2024.11.6)
Requirement already satisfied: requests~=2.26 in ./venv/lib64/python3.11/site-packages (from mkdocs-material->-r requirements.txt (line 2)) (2.32.3)
Requirement already satisfied: python-dateutil>=2.8.1 in ./venv/lib64/python3.11/site-packages (from ghp-import>=1.0->mkdocs->-r requirements.txt (line 1)) (2.9.0.post0)
Requirement already satisfied: platformdirs>=2.2.0 in ./venv/lib64/python3.11/site-packages (from mkdocs-get-deps>=0.2.0->mkdocs->-r requirements.txt (line 1)) (4.3.6)
Requirement already satisfied: charset-normalizer<4,>=2 in ./venv/lib64/python3.11/site-packages (from requests~=2.26->mkdocs-material->-r requirements.txt (line 2)) (3.4.0)
Requirement already satisfied: idna<4,>=2.5 in ./venv/lib64/python3.11/site-packages (from requests~=2.26->mkdocs-material->-r requirements.txt (line 2)) (3.10)
Requirement already satisfied: urllib3<3,>=1.21.1 in ./venv/lib64/python3.11/site-packages (from requests~=2.26->mkdocs-material->-r requirements.txt (line 2)) (2.2.3)
Requirement already satisfied: certifi>=2017.4.17 in ./venv/lib64/python3.11/site-packages (from requests~=2.26->mkdocs-material->-r requirements.txt (line 2)) (2024.8.30)
Requirement already satisfied: six>=1.5 in ./venv/lib64/python3.11/site-packages (from python-dateutil>=2.8.1->ghp-import>=1.0->mkdocs->-r requirements.txt (line 1)) (1.17.0)
/[~/kubesaw/kubesaw.github.io] ((06c2bc5...))$ mkdocs serve
Error: Config file 'mkdocs.yml' does not exist.
```

Reply:
I think you did everything correctly. The problem is that the `mkdocs.yml` file was added in this PR https://github.com/kubesaw/kubesaw.github.io/pull/5 and not the current one 🤦‍♂️

Reply:
But now that I've merged that, I'm noticing that GH doesn't suggest me to update this PR 🤔 . I'm wondering why.

Reply:
Anyways , I've rebased the PR and tried again, it should be working now. Sorry for the confusion. Please let me know if it works for you.

As per the PR update button probably I'm missing some configuration in project.

Reply:
> But now that I've merged that, I'm noticing that GH doesn't suggest me to update this PR 🤔 . I'm wondering why.

it's in the general settings of the project - I updated it.
Btw, it was also missing the branch protection for the `main` branch - I added it so it requires at least one review approval.

Reply:
thanks a lot for fixing it 🙏